Source/WPEFramework/PluginServer.h

```diff
@@ -3609,37 +3609,61 @@ POP_WARNING()
+                                    response->ErrorCode = Web::STATUS_NO_CONTENT;
+                                    message->Clear();
+                                    message->Id = id;
+                                    message->Error.SetError(Core::ERROR_INVALID_INPUT_LENGTH);
+                                    message->Error.Text = error;
+                                    response->Body(Core::ProxyType<Web::IBody>(message));
+                                }
                             }
+                            else {
+                                if (HasService() == true) {
+                                    message->ImplicitCallsign(GetService().Callsign());
+                                }
 
-                            if (message->IsSet()) {
-                                Core::ProxyType<Core::JSONRPC::Message> body = Job::Process(_token, message);
+                                if (message->IsSet()) {
+                                    Core::ProxyType<Core::JSONRPC::Message> body = Job::Process(_token, Core::ProxyType<Core::JSONRPC::Message>(message));
 
-                                // If we have no response body, it looks like an async-call...
-                                if (body.IsValid() == false) {
-                                    // It's a a-synchronous call, seems we should just queue this request, it will be answered later on..
-                                    if (_request->Connection.Value() == Web::Request::CONNECTION_CLOSE) {
-                                        Job::RequestClose();
+                                    // If we have no response body, it looks like an async-call...
+                                    if (body.IsValid() == false) {
+                                        // It's a a-synchronous call, seems we should just queue this request, it will be answered later on..
+                                        if (_request->Connection.Value() == Web::Request::CONNECTION_CLOSE) {
+                                            Job::RequestClose();
+                                        }
+                                    }
+                                    else {
+                                        response = IFactories::Instance().Response();
+                                        response->Body(body);
+                                        if (body->Error.IsSet() == false) {
+                                            response->ErrorCode = Web::STATUS_OK;
+                                            response->Message = _T("JSONRPC executed succesfully");
+                                        }
+                                        else {
+                                            response->ErrorCode = Web::STATUS_ACCEPTED;
+                                            response->Message = _T("Failure on JSONRPC: ") + Core::NumberType<uint32_t>(body->Error.Code).Text();
```

Comment:
Error code here can be negative if it's a JSON-RPC error so will overflow uint32_t

Reply:
Correct, will adapt to int32_t